File: cns/R/srm/snnSRM/src/grad_funcs.cpp
#include "grad_funcs.h"

double p_stroke(const double &t, const SInput &si) {
 //printf("beta: %f, alpha: %f, tr: %f, u(t,si) = %f\n", (double)si.c["beta"], (double)si.c["alpha"], (double)si.c["tr"], u(t, si));
 return si.get_c("beta")/(1+exp(si.get_c("alpha")*(si.get_c("tr")-u(t, si))));
}

double grab_epsp(const double &t, const SSynInput &ssyn) {
  double e_syn = 0;
  const NumericVector &sp(ssyn.si.net[ssyn.syn_id-1]);
  for(int syn_id=sp.size()-1; syn_id>=0; syn_id--) {
    if(t - sp[syn_id] > EPSP_WORK_WINDOW) break;
    e_syn += epsp(t - sp[syn_id], ssyn.si.c);
  }
  return e_syn;
}

File: cns/R/srm/grad_funcs.R


grab_epsp <- Vectorize(function(t, sp) {
    sum(epsp(t-sp))
},"t")

  

p_stroke <- Vectorize(function(u) {
    beta/(1+ exp(alpha*(tr-u))) 
})


ratecalc = function(w) {
  w4 = w^6
  0.04* w4/(w4+ws^6) 
}

#ww = seq(0,20, by=0.1)
#plot(ww, ratecalc(ww), type="l")